### PR TITLE
[REF] Extract domain tokens to their own class

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Note that the moved function was only extracted a few days ago.

Overview
----------------------------------------
[REF] Extract domain tokens to their own class 

Before
----------------------------------------
Domain tokens are part of compatSubscriber

After
----------------------------------------
Their very own class

Technical Details
----------------------------------------
Note the moved function is a very recent extraction so won't be called from anywhere else

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Core/DomainTokens.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+   *
+   * @param int|null $domainID
+   * @param bool $html
+   *
+   * @return array
+   * @throws \CRM_Core_Exception
+   * @internal
+   *
+   */
+  public static function getDomainTokenValues(?int $domainID = NULL, bool $html = TRUE): array {
```
Comment (eileenmcnaughton) [inline CRM/Core/DomainTokens.php:75]: this function is moved but otherwise unchanged by this PR - it was recently extracted from a bigger function

**Comment by eileenmcnaughton** [comment]
@totten this is now passing - most of the changes are in the deprecated functions to get the token list - so they are a bit cludgey but have limited life ahead of them.

Once this is merged I'll add a `{domain.now}` token and use that to write the tests for the smarty functions.

I'd also like to replace all calls to `replaceDomainTokens` - I think that change would be pretty simple. I'm wanting to deprecate most of the functions on `CRM_Utils_Token`

**Comment by colemanw** [comment]
Refactoring looks good.

**Comment by eileenmcnaughton** [comment]
thanks @colemanw @mlutfy

**Pull request merged**